### PR TITLE
Feature/3.0.0/vp 828 implement client credential flow

### PR DESCRIPTION
**Pull request by anivirtoway**
for https://virtocommerce.atlassian.net/browse/VP-828

### CODE REVIEW

File: VirtoCommerce.Storefront/Infrastructure/Autorest/ClientCredentialAuthHandler.cs [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+using System.Net.Http.Headers;
+using System.Threading;
+using System.Threading.Tasks;
+using IdentityModel.Client;
+using Microsoft.Extensions.Options;
+
+namespace VirtoCommerce.Storefront.Infrastructure.Autorest
+{
+    /// <summary>
+    /// HTTP message delegating handler that encapsulates access token handling and renewment
```
Comment (artem-dudarev) [inline VirtoCommerce.Storefront/Infrastructure/Autorest/ClientCredentialAuthHandler.cs]: renewal

File: VirtoCommerce.Storefront/Infrastructure/Autorest/ClientCredentialAuthHandler.cs [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    /// </summary>
+    public class ClientCredentialAuthHandler : DelegatingHandler
```
Comment (artem-dudarev) [inline VirtoCommerce.Storefront/Infrastructure/Autorest/ClientCredentialAuthHandler.cs]: ClientCredentialsAuthHandler

File: VirtoCommerce.Storefront/Infrastructure/Autorest/ClientCredentialAuthHandler.cs [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+        /// Sends an HTTP request to the inner handler to send to the server as an asynchronous operation.
+        /// </summary>
+        /// <param name="request">The HTTP request message to send to the server.</param>
+        /// <param name="cancellationToken">A cancellation token to cancel operation.</param>
+        /// <returns>
+        /// Returns <see cref="T:System.Threading.Tasks.Task`1" />. The task object representing the asynchronous operation.
+        /// </returns>
+        protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+        {
+
```
Comment (artem-dudarev) [inline VirtoCommerce.Storefront/Infrastructure/Autorest/ClientCredentialAuthHandler.cs]: Remove

File: VirtoCommerce.Storefront/Infrastructure/Autorest/ClientCredentialAuthHandler.cs (hunk 1 above)
Comment (anivirtoway) [reply VirtoCommerce.Storefront/Infrastructure/Autorest/ClientCredentialAuthHandler.cs]: done

**Pull request merged**